### ISSUE TITLE
Bocfel: glk_window_move_cursor call sets wrong cursor position

### ISSUE THREAD

**Issue by angstsmurf** [garglk/garglk#383]
The call to glk_window_move_cursor() in Bocfel at[ line 326 in screen.c](https://github.com/garglk/garglk/blob/master/terps/bocfel/screen.c#L326) seems to move the cursor, i.e. the current printing position in the status window, an extra `input->len` number of characters to the right. 
This breaks the Function Key Definitions functionality in Beyond Zork.

To reproduce, type DEFINE as a command in Beyond Zork. Then press any function key. Duplicate text is printed (and line input started) at the wrong position.

Removing the line seems to fix the problem without any obvious adverse effects.

EDIT: Actually, removing [line 318](https://github.com/garglk/garglk/blob/master/terps/bocfel/screen.c#L318) instead seems to be a better idea. After implementing mouse support, the other way caused some issues.

**Comment by cspiegel** [comment]
Yeesh, this is an ugly one.  Simply removing the line doesn't work because the original code is correct in the case of no preloaded input _but_ input was terminated by a terminating key.  The problem is that preloaded text is printed, resulting in `input->x` being advanced already, thus the doubling.  It looks like the following change to line 318 works, if I'm understanding all the interactions properly:

```c
if(input->term != ZSCII_NEWLINE) upperwin->x += input->len - input->preloaded;
```

But you've also uncovered another problem, as the preloaded text in the DEFINE menu is broken (giving both a lower- and uppercase version of the text).  This is ridiculous because it used to work, but I fixed a similar problem for preloaded text at the command prompt: in previous versions, hitting a function key would result in something like:

`EXAMINEexamine`

Now in the DEFINE menu, you get:

`examineEXAMINE`

The cases are swapped, and I used the knowledge that BZ was preloading input with lowercase to hack around the problem.  But for the DEFINE menu it's preloading in uppercase, but printing lower!  The exact opposite problem, meaning the hack doesn't work there.

I think I'll just add a case-insensitive "unput" function to Gargoyle instead of trying to work around this inside of Bocfel.
